files/brews/hub.rb

```diff
@@ -0,0 +1,26 @@
+require 'formula'
+
+class Hub < Formula
+  homepage "http://hub.github.com/"
+
+  if RUBY_PLATFORM =~ /x86_64/
+    url "https://github.com/github/hub/releases/download/v2.2.0-rc1/hub_mac_amd64_v2.2.0-rc1.tar.gz"
+    sha1 "413ce80d32f6313cef010e0edb1e664e1fff7427"
+  else
+    url "https://github.com/github/hub/releases/download/v2.2.0-rc1/hub_mac_386_v2.2.0-rc1.tar.gz"
+    sha1 "059e6d1759b568fd79bacfc397de199135056ff1"
+  end
+
+  def install
+    bin.install "hub"
+    man1.install Dir["man/*"]
+    bash_completion.install "etc/hub.bash_completion.sh"
+    zsh_completion.install "etc/hub.zsh_completion" => "_hub"
+  end
+
+  test do
+    HOMEBREW_REPOSITORY.cd do
+      assert_equal "bin/brew", shell_output("#{bin}/hub ls-files -- bin").strip
```

Comment:
I strongly disagree with that advice. /cc @mikemcquaid

If Homebrew changes its internal structure around `bin/`, this test will start failing and it wouldn't be our fault. That makes it brittle.

On the other hand, checking for version is a solid test that verifies that hub works and is the actual exact version installed by this formula.

We can't have hub hit the GitHub API because that would force users to go through the interactive authentication dance if the user hasn't done so.